### PR TITLE
[JBIDE-21424] fix parsing host from url

### PR DESCRIPTION
**Pull request by fbricon**
Signed-off-by: Fred Bricon <fbricon@gmail.com>

### CODE REVIEW

File: plugins/org.jboss.tools.openshift.common.core/src/org/jboss/tools/openshift/common/core/utils/UrlUtils.java [hunk 1]
```diff
@@ -163,25 +167,28 @@ public static boolean hasScheme(String host) {
...
+			} catch (URISyntaxException ignored) {
+			}
 		}
-		
-		int endIndex = url.length() - 1;
-		if (url.endsWith("/")) {
-			endIndex -= 1;
+
+		if (host == null) {
+			Matcher m = SIMPLE_QUASI_URL_PATTERN.matcher(url);
```
Comment (adietish) [inline plugins/org.jboss.tools.openshift.common.core/src/org/jboss/tools/openshift/common/core/utils/UrlUtils.java]: what's the difference btw. SIMPLE_QUASI_URL_PATTERN and SIMPLE_URL_PATTERN? Cant we reuse?

Comment (fbricon) [inline plugins/org.jboss.tools.openshift.common.core/src/org/jboss/tools/openshift/common/core/utils/UrlUtils.java]: SIMPLE_QUASI_URL_PATTERN handles non-urls

Comment (adietish) [inline plugins/org.jboss.tools.openshift.common.core/src/org/jboss/tools/openshift/common/core/utils/UrlUtils.java]: +1 LGTM

**Comment by fbricon** [comment]
Applies in master and 4.3.x

**Pull request closed without merging**